include/openmc/partitioner_utils.h

```diff
@@ -0,0 +1,322 @@
+#ifndef OPENMC_PARTITIONER_UTILS_H
+#define OPENMC_PARTITIONER_UTILS_H
+
+#include "position.h"
+#include "random_dist.h"
+#include "timer.h"
+#include "universe.h"
+
+#include <omp.h>
+#include <set>
+#include <vector>
+
+namespace openmc {
+
+using vec3 = Position;
+
+// This axis-aligned bounding box class (AABB) is designed to work easier with
+// partitioners than openmc::BoundingBox
+struct AABB {
+  AABB();
+  AABB(const vec3& mi, const vec3& ma);
+
+  void extend_max(const vec3& val);
+  void extend_min(const vec3& val);
+
+  void extend(const vec3& pos);
+  void extend(const AABB& other_box);
+
+  vec3 get_center() const;
+  float surface_area() const;
```

Comment:
I switched to double precision in most areas. In some areas, I kept single precision for better cache efficiency. 

I think it would be nice to turn `Position` into a template, where `Postion<double>` means you want double precision (the default if no type is supplied), `Position<float>` means you want single precision, etc. It would help in certain areas (e.g. the kd tree partitioner) where double precision is not required and memory/cache efficiency is a bigger concern.